pxr/usd/lib/usd/stage.cpp

```diff
@@ -3154,12 +3154,24 @@ UsdStage::_DefinePrim(const SdfPath &path, const TfToken &typeName)
 {
     // Special-case requests for the root.  It always succeeds and never does
     // authoring since the root cannot have PrimSpecs.
-    if (path == SdfPath::AbsoluteRootPath())
+    if (path == SdfPath::AbsoluteRootPath()) {
         return GetPseudoRoot();
+    }
 
     // Define all ancestors.
-    if (!_DefinePrim(path.GetParentPath(), TfToken()))
+    auto parentPath = path.GetParentPath();
+    if (!_DefinePrim(parentPath, TfToken())) {
         return UsdPrim();
+    }
+
+    // Disallow authoring a prim under an inactive ancestor
+    if (!GetPrimAtPath(parentPath).IsActive()) {
```

Comment:
@superfunc - If path is more than one level deep beneath an inactive prim, its *parent* prim won't even exist.  So at a minimum, you should also be checking the result of GetPrimAtPath(), first.  You could go further, and, if that fails, walk up the ancestor paths to find the actually inactive prim to report in your error, which I *think* should be a TF_WARN rather than a RUNTIME error.  The error below is RUNTIME because we have no clue why our action failed, so it's a "system failure".  In this case, though, we know what the problem was: you tried to do something you shouldn't have.

Thanks - this is a nice addition!
--spiff

Reply:
Ah, you're totally right, @superfunc - I didn't expand out the code in the review and so didn't notice it was *self* recursion!  It would be good to clarify in the new comment at line 3167, though, that due to the recursion we are guaranteed that there will be a prim at parentPath (though the paranoid in me still wants you to check the prim first before calling IsActive() :-)